SLING-12737 The method findJobs in JobManagerImpl does not accept a template containing an operation

mirsilstan opened the pull request:
https://issues.apache.org/jira/browse/SLING-12737

The property name is now encoded only after extracting the operation from the first 1 or 2 character of the key.
I added a test in HistoryIT that retrieves the jobs between a start date and an end date.

joerghoh commented on src/main/java/org/apache/sling/event/impl/support/PropertyNameAndOperationExtractor.java:
```diff
@@ -0,0 +1,37 @@
+package org.apache.sling.event.impl.support;
+
+public class PropertyNameAndOperationExtractor {
+    public PropertyNameAndOperation extractPropertyNameAndOperation(String key) {
```
I would convert this into a static method, then you won't need to invoke the constructor anymore.

joerghoh commented:
@mirsilstan thanks for your contribution.

If I understand your description in SLING-12737, the code in ``JobSchedulerImpl`` is correct and would be able to handle the templates as intended, but not in the ``JobManagerImpl``; is my understanding correct?

I would appreciate a bit of test coverage for the code you extracted into the ``PropertyNameAndOperationExtractor``class.

mirsilstan commented:
> @mirsilstan thanks for your contribution.
> 
> If I understand your description in [SLING-12737](https://issues.apache.org/jira/browse/SLING-12737), the code in `JobSchedulerImpl` is correct and would be able to handle the templates as intended, but not in the `JobManagerImpl`; is my understanding correct?
> 
> I would appreciate a bit of test coverage for the code you extracted into the `PropertyNameAndOperationExtractor`class.

The only reason I added the extra classes is because the code in JobManagerImpl and the code in JobSchedulerImpl is duplicated.
I thought about creating a single method and reusing it in both places.
However, this increases the number of changes.
Hence I removed the extra classes and I changed only 2 lines which are strictly related to the bug.
The code is tested in HistoryIT, if you run the test on the main branch, it will fail.
Please advise if any further tests are necessary.
